src/main/java/marquez/db/DatasetDao.java

```diff
@@ -144,9 +143,44 @@ void updateLastModifedAt(
           + "WHERE n.name = :namespaceName "
           + "ORDER BY d.name "
           + "LIMIT :limit OFFSET :offset")
-  @RegisterRowMapper(ExtendedDatasetRowMapper.class)
   List<ExtendedDatasetRow> findAll(String namespaceName, int limit, int offset);
 
   @SqlQuery("SELECT COUNT(*) FROM datasets")
   int count();
+
+  @SqlQuery(
+      "INSERT INTO datasets ("
+          + "type, "
+          + "created_at, "
+          + "updated_at, "
+          + "namespace_uuid, "
+          + "source_uuid, "
+          + "name, "
+          + "physical_name, "
+          + "description "
+          + ") VALUES ( "
+          + ":type, "
+          + ":now, "
+          + ":now, "
+          + ":namespaceUuid, "
+          + ":sourceUuid, "
+          + ":name, "
+          + ":name, "
+          + ":description) "
+          + "ON CONFLICT (namespace_uuid, name) "
+          // todo: what is the unique identifier here?
```

Comment:
A dataset is unique on `namespace_uuid`, `source_uuid`, `name` and `physical_name` (see [initial schema](https://github.com/MarquezProject/marquez/blob/main/src/main/resources/marquez/db/migration/V1__initial_schema.sql#L47))